p2p/net/upgrader/upgrader.go

```diff
@@ -177,6 +178,8 @@ func (u *upgrader) upgrade(ctx context.Context, t transport.Transport, maconn ma
 		return nil, fmt.Errorf("failed to negotiate stream multiplexer: %s", err)
 	}
 
+	canonicallog.LogPeerStatus(100, sconn.RemotePeer(), maconn.RemoteMultiaddr(), "connection_status", "established")
```

Comment:
Should this happen in the swarm, so we also catch QUIC connections?

Reply:
I added a log to quic transport directly. Would you prefer I do this in the swarm?

Reply:
Yes, that way we'd make sure that every connection is logged, even if it doesn't use the upgrader. We'll thank ourselves for that when we add WebTransport.

You can do it in a follow-up PR, if you prefer.